### PR TITLE
fix(cozy-search): Add fallback message for empty assistant replies

### PR DESCRIPTION
**Pull request by ewan102**
When the RAG backend returns an empty or citation-only answer, display a translated fallback message instead of a blank bubble. Applied consistently across the live streaming path, conversation history reload, and the conversation list preview.

## Summary by CodeRabbit

* **Bug Fixes**
  * Added a localized fallback message when assistant responses are empty or contain only sanitized content.
  * Conversation previews now display the fallback message when no usable assistant response is available.
  * Improved handling of streamed responses to prevent blank answers.

* **Localization**
  * Added empty-response translations in English, French, Russian, and Vietnamese.

### CODE REVIEW

File: packages/cozy-search/src/components/adapters/CozyRealtimeChatAdapter.ts [hunk 1]
```diff
@@ -129,8 +129,15 @@ export const createCozyRealtimeChatAdapter = (
       if (!wasAborted) {
         const finalText = sanitizeChatContent(fullText)
         const sources = streamBridge.getSources(conversationId)
+        if (!finalText.trim()) {
+          log.info(
+            `Empty content was received for conversation ${conversationId}, using fallback message`
+          )
+        }
```
Comment (zatteo) [inline packages/cozy-search/src/components/adapters/CozyRealtimeChatAdapter.ts]: If you really want to log when the fallback message is displayed, you should do it directly in `withEmptyResponseFallback` function to avoid checking 2x with `trim()` no?

File: packages/cozy-search/src/components/CozyAssistantRuntimeProvider.tsx [hunk 2]
```diff
@@ -73,19 +73,28 @@ interface CozyAssistantRuntimeProviderProps {
...
-      msg.role === 'assistant' && msg.sources
-        ? { custom: { sources: msg.sources } }
-        : undefined
-  }))
+  return messages.map((msg, idx) => {
+    const sanitized = sanitizeChatContent(msg.content)
+    const content =
+      msg.role === 'assistant'
+        ? withEmptyResponseFallback(sanitized, t)
+        : sanitized
```
Comment (zatteo) [inline packages/cozy-search/src/components/CozyAssistantRuntimeProvider.tsx]: It seems that we have multiple time this pattern. Maybe we should create a `formatAnswer` function that does everything? cc @paultranvan

Comment (paultranvan) [reply packages/cozy-search/src/components/CozyAssistantRuntimeProvider.tsx]: sounds like a good idea

File: packages/cozy-search/src/components/adapters/CozyRealtimeChatAdapter.ts (hunk 1 above)
Comment (paultranvan) [reply packages/cozy-search/src/components/adapters/CozyRealtimeChatAdapter.ts]: I'm not sure it's worth having this log, tough

File: packages/cozy-search/src/components/helpers.js [hunk 3]
```diff
@@ -85,8 +98,11 @@ export const getNameOfConversation = conversation => {
  * Since we don't have rule for description of the conversation
  * So temporary we get the last answer from assistant as description of the conversation
  */
-export const getDescriptionOfConversation = conversation => {
-  const content =
-    conversation.messages?.[conversation.messages.length - 1]?.content
-  return content && sanitizeChatContent(content)
+export const getDescriptionOfConversation = (conversation, t) => {
+  const lastMessage = conversation.messages?.[conversation.messages.length - 1]
```
Comment (paultranvan) [inline packages/cozy-search/src/components/helpers.js]: we should have a first check on conversation in case it's null / undefined

**Review by paultranvan — COMMENTED** [review]
I'm thinking of an edge case that I'm unsure of the behaviour: 
- user sends query
- the user stops the generation before the first token is received
- is the fallback displayed in this case? I think it shouldn't
Can you test this case @ewan102 ?

File: packages/cozy-search/src/components/adapters/CozyRealtimeChatAdapter.ts (hunk 1 above)
Comment (ewan102) [reply packages/cozy-search/src/components/adapters/CozyRealtimeChatAdapter.ts]: Ok so I get rid of the log or do I keep it and put it in with EmptyResponseFallback ?

**Comment by ewan102** [comment]
Tested @paultranvan 's case, it does not put the fallback answer in that case. Got rid of the log info

File: packages/cozy-search/src/components/helpers.js (hunk 3 above)
Comment (paultranvan) [reply packages/cozy-search/src/components/helpers.js]: @ewan102 Thank you for the check ; note you could have just added a `?` rather than a `if(!conversation)` : 
`  const lastMessage = conversation?.messages?.[conversation.messages.length - 1]`

File: packages/cozy-search/src/components/adapters/CozyRealtimeChatAdapter.ts (hunk 1 above)
Comment (paultranvan) [reply packages/cozy-search/src/components/adapters/CozyRealtimeChatAdapter.ts]: You can remove it
